File: Leetcode/268. Missing Number/268_Missing_Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += nums[i];
        }
        return (n * (n + 1) / 2) - sum;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = { 3, 0, 1 };
        System.out.println(solution.missingNumber(nums));
    }
}

File: Leetcode/268. Missing Number/268_Missing_Number.py
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        return n * (n + 1) // 2 - sum(nums)


class Solution2:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        res = n
        for i in range(n):
            res ^= i ^ nums[i]
        return res


class Main:
    def execute(self):
        solution = Solution()
        nums = [3, 0, 1]
        print(solution.missingNumber(nums))

        solution2 = Solution2()
        print(solution2.missingNumber(nums))


main = Main()
main.execute()
